src/OpenLoco/src/GameSaveCompare.cpp

```diff
@@ -0,0 +1,508 @@
+#include "GameSaveCompare.h"
+
+#include <string>
+
+#include "Effects/EffectsManager.h"
+#include "GameState.h"
+#include "Logging.h"
+#include "OpenLoco.h"
+#include "S5/Limits.h"
+#include "S5/S5.h"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Diagnostics;
+
+namespace OpenLoco::GameSaveCompare
+{
+    std::string getVehicleSubType(const Vehicles::VehicleEntityType vehicleSubType);
+    std::string getEffectSubType(const EffectType effectSubType);
+    void logVehicleTypeAndSubType(int offset, const OpenLoco::Entity& entity);
+    void logEffectType(int offset, const OpenLoco::Entity& entity);
+    long logDivergentEntityOffset(const OpenLoco::Entity& lhs, const S5::Entity& rhs, int offset, bool displayAllDivergences, long divergentBytesTotal);
+    bool compareGameStates(GameState& gameState1, S5::GameState& gameState2, bool displayAllDivergences);
+    bool logDivergenceRoutings(OpenLoco::GameState& gameState1, OpenLoco::S5::GameState& gameState2, bool displayAllDivergences);
+    bool compareElements(const std::vector<S5::TileElement>& tileElements1, const std::vector<S5::TileElement>& tileElements2, bool displayAllDivergences);
+
+    template<typename T>
+    std::span<const std::byte> getBytesSpan(const T& item)
+    {
+        return std::span<const std::byte, sizeof(T)>{ reinterpret_cast<const std::byte*>(std::addressof(item)), sizeof(T) };
```

Comment:
I don't think that we need to have the size in the template arguments, this will most likely construct a temporary and creates a new span object as the resulting type does not encode the size.